tools/utils/array_readwriter/numpy_array.py

```diff
@@ -0,0 +1,23 @@
+import logging
+import numpy as np
+from numpy.lib.format import open_memmap
+from .registry import register
+
+@register("numpy")
+class Numpy(object):
```

Comment:
i wonder why we need this wrapper? where is it used?
what is the benefit of doing this? the code should be intuitive. it's hard for me to see how this code works. i'm not sure if this is easy for others to understand.

Reply:
I guess it's for reflection.
E.g.,
```
"format" : {"name": "csv", "delimiter": " "},
```
should use the `CSV` parser class, while for
```
"format" : {"name": "numpy"},
```
should use the `Numpy` parser class.

However, the code is not very readable, my suggestions:
- Use name like `NumpyArrayParser` instead of `Numpy`.
- Use `register_array_parser` instead of `register`.
- For getting the parser class from the name,
```python
from .array_readwriter import get_parser
parser = get_parser(name='numpy', **other_kwargs)
parser.read(...)
parser.write(...)
```
Currently, it is hided in the `array_readwriter` module. It is usually difficult for ppl to realize that the reflection is done implicit.

Reply:
@zheng-da The benefit is not only what @jermainwang mentioned, but also that it will be straightforward to add in new formats later (either by us or by other people who are modifying our code).  One just needs to write a new class in a new Python file and register it instead of modifying an if-else block or a dictionary.  I'm open to other patterns so let me know if you have other suggestions.

Reply:
i see what you are doing here. i think i understand the code now.